source/platforms/snowflake.md

```diff
@@ -0,0 +1,343 @@
+
+First we login into the snowflake repository with docker, via terminal:
+
+```{note}
+If you have **2FA** both `docker login` and `docker push` commands don't
+work as expected. Every API call results in a push notification to approve, but
+if not done promptly teh original API times out, complicating the completion of
+this step.
+
+We recommend enabling [token caching](https://docs.snowflake.com/en/user-guide/security-mfa#using-mfa-token-caching-to-minimize-the-number-of-prompts-during-authentication-optional)
+or disabling 2FA during this step.
+```
+
+```bash
+docker login <snowflake_registry_hostname> -u <snowflake_user_name>
+> prompt for password
+```
+
+We tag and push the image, make sure you replace the repository url for `org-account.registry.snowflakecomputing.com/container_hol_db/public/image_repo`:
+
+```bash
+docker tag <local_repository>/rapids-nb-snowflake:latest <repository_url>/rapids-nb-snowflake:dev
+```
+
+Verify that the new tagged image exists by running:
+
+```bash
+docker image list
+```
+
+Push the image to snowflake:
+
+```bash
+docker push <repository_url>/rapids-nb-snowflake:dev
+```
+
+```{note}
+This step will take some time, while this process completes we can continue
+with next step to configure and push the Spec YAML.
+```
+
+When the `docker push` command completes, you can verify that the image exists in your Snowflake Image Repository by running the following in the Snowflake SQL worksheet
+
+```{code-block} sql
+:force:
+
+USE ROLE CONTAINER_USER_ROLE;
+CALL SYSTEM$REGISTRY_LIST_IMAGES('/CONTAINER_HOL_DB/PUBLIC/IMAGE_REPO');
+```
+
+## Configure and Push Spec YAML
+
+Snowpark Container Services are defined and configured using YAML files. There
+is support for multiple parameters configurations, refer to then [Snowpark container services specification reference](https://docs.snowflake.com/en/developer-guide/snowpark-container-services/specification-reference) for more information.
+
+Locally, create the following file `rapids-snowpark.yaml`:
+
+```yaml
+spec:
+  containers:
+    - name: rapids-nb-snowpark
+      image: <org-account>.registry.snowflakecomputing.com/container_hol_db/public/image_repo/rapids-nb-snowflake:dev
+      volumeMounts:
+        - name: rapids-notebooks
+          mountPath: /home/rapids/notebooks/workspace
+      resources:
+        requests:
+          nvidia.com/gpu: 1
+        limits:
+          nvidia.com/gpu: 1
+  endpoints:
+    - name: jupyter
+      port: 8888
+      public: true
+    - name: dask-client
+      port: 8786
+      protocol: TCP
+      public: true
+    - name: dask-dashboard
+      port: 8787
+      public: true
+  volumes:
+    - name: rapids-notebooks
+      source: "@volumes/rapids-notebooks"
+      uid: 1001 # rapids user's UID
+      gid: 1000
```

Comment:
I tested this persistence out by creating a notebook and deleting and recreating the service and it worked great 🎉 

<img width="723" alt="image" src="https://github.com/user-attachments/assets/0677831e-234f-44c7-9237-46ef0faf9962" />

Reply:
Great! Yes I tested it too I should have mentioned that : )